mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresMailboxAnnotationDAO.java

```diff
@@ -0,0 +1,77 @@
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.postgres.mail.dao;
+
+import static org.apache.james.mailbox.postgres.PostgresMailboxAnnotationModule.PostgresMailboxAnnotationTable.ANNOTATIONS;
+import static org.apache.james.mailbox.postgres.PostgresMailboxAnnotationModule.PostgresMailboxAnnotationTable.MAILBOX_ID;
+import static org.apache.james.mailbox.postgres.PostgresMailboxAnnotationModule.PostgresMailboxAnnotationTable.TABLE_NAME;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.apache.james.mailbox.model.MailboxAnnotationKey;
+import org.apache.james.mailbox.postgres.PostgresMailboxId;
+import org.jooq.postgres.extensions.types.Hstore;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class PostgresMailboxAnnotationDAO {
+    private final PostgresExecutor postgresExecutor;
+
+    public PostgresMailboxAnnotationDAO(PostgresExecutor postgresExecutor) {
+        this.postgresExecutor = postgresExecutor;
+    }
+
+    public Flux<MailboxAnnotation> getAllAnnotations(PostgresMailboxId mailboxId) {
+        return postgresExecutor.executeRows(dslContext ->
+            Flux.from(dslContext.selectFrom(TABLE_NAME)
+                .where(MAILBOX_ID.eq(mailboxId.asUuid()))))
+            .singleOrEmpty()
+            .map(record -> record.get(ANNOTATIONS))
+            .flatMapIterable(this::hstoreToAnnotations);
+    }
+
+    private List<MailboxAnnotation> hstoreToAnnotations(Hstore hstore) {
+        return hstore.data()
+            .entrySet()
+            .stream()
+            .map(entry -> MailboxAnnotation.newInstance(new MailboxAnnotationKey(entry.getKey()), entry.getValue()))
+            .collect(Collectors.toList());
+    }
+
+    public Mono<Void> insertAnnotation(PostgresMailboxId mailboxId, MailboxAnnotation mailboxAnnotation) {
+        Preconditions.checkArgument(!mailboxAnnotation.isNil());
+
+        // TODO: likely wrong... might need to check if row exists with mailboxId and then insert if not, update if is
+        return postgresExecutor.executeVoid(dslContext ->
+            Mono.from(dslContext.insertInto(TABLE_NAME, MAILBOX_ID, ANNOTATIONS)
+                .values(mailboxId.asUuid(), annotationAsHstore(mailboxAnnotation))));
```

Comment:
Here we use HStore on a mutable field. 
Thus we likely need in place updates.

```
UPDATE votre_table
SET votre_colonne_hstore = votre_colonne_hstore || 'nouvelle_cle' => 'nouvelle_valeur'
WHERE votre_condition;

UPDATE votre_table
SET votre_colonne_hstore = votre_colonne_hstore - 'cle_a_retirer'
WHERE votre_condition;
```

```
dsl.update(table("votre_table"))
   .set(field("votre_colonne_hstore"), field("votre_colonne_hstore").concat(field("'nouvelle_cle => nouvelle_valeur'")))
   .where(votre_condition)
   .execute();
   
dsl.update(table("votre_table"))
   .set(field("votre_colonne_hstore"), field("votre_colonne_hstore").minus(field("'cle_a_retirer'")))
   .where(votre_condition)
   .execute();
```

Reply:
Not only cause we have the case when the line with mailboxId does not exist yet, so we need to do an insert. Hung and Quan did something like that with quota DAOs, and I intend to reuse something similar